atomics/include/desul/atomics/Compare_Exchange_ScopeCaller.hpp

```diff
@@ -8,24 +8,23 @@ SPDX-License-Identifier: (BSD-3-Clause)
 
 #ifndef DESUL_ATOMICS_COMPARE_EXCHANGE_SCOPECALLER_HPP_
 #define DESUL_ATOMICS_COMPARE_EXCHANGE_SCOPECALLER_HPP_
-#include "desul/atomics/Common.hpp"
 
-namespace desul {
+#include <desul/atomics/Common.hpp>
 
-template <class MemoryOrder>
-DESUL_INLINE_FUNCTION void atomic_thread_fence(MemoryOrder, MemoryScopeCaller) {}
+namespace desul {
+namespace Impl {
 
 #define DESUL_ATOMIC_EXCHANGE_SCOPECALLER(MEMORY_ORDER)               \
-  template <typename T>                                               \
-  DESUL_INLINE_FUNCTION T atomic_exchange(                            \
+  template <class T>                                                  \
+  DESUL_INLINE_FUNCTION T host_atomic_exchange(                       \
       T* dest, T value, MEMORY_ORDER, MemoryScopeCaller) {            \
     T return_val = *dest;                                             \
     *dest = value;                                                    \
     return return_val;                                                \
   }                                                                   \
                                                                       \
-  template <typename T>                                               \
-  DESUL_INLINE_FUNCTION T atomic_compare_exchange(                    \
+  template <class T>                                                  \
+  DESUL_INLINE_FUNCTION T host_atomic_compare_exchange(               \
```

Comment:
doesn't this need a device and a host overload?

Reply:
Yes you're right

Reply:
I went ahead and defined an `atomic_compare_exchange` overload directly (shortcutting the `host_` and `device_` version.